packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx

```diff
@@ -172,7 +173,7 @@ export const MultiSectionDigitalClockSection = React.forwardRef(
       const activeItem = containerRef.current.querySelector<HTMLElement>(
         '[role="option"][tabindex="0"], [role="option"][aria-selected="true"]',
       );
-      if (active && autoFocus && activeItem) {
+      if (activeItem && active && autoFocus && activeItem !== previousActive.current) {
```

Comment:
Fix the issue of stealing the focus when closing a picker with MultiSectionDigitalClock:

https://github.com/mui/mui-x/assets/4941090/7fbec835-f991-4831-99f4-2e82fd4f3de0